server/src/main/java/io/druid/server/initialization/jetty/JettyServerModule.java

```diff
@@ -161,6 +162,10 @@ private static Server makeJettyServer(@Self DruidNode node, ServerConfig config)
 
     final Server server = new Server(threadPool);
 
+    // Without this bean set, the default ScheduledExecutorScheduler runs as non-daemon, causing lifecycle hooks to fail
+    // to fire on main exit. Related bug: https://github.com/druid-io/druid/pull/1627
+    server.addBean(new ScheduledExecutorScheduler("JettyScheduler", true), true);
```

Comment:
Stopping the life cycle will still be done explicitly until there are at least better integration tests in place